mobile/src/main/java/org/openhab/habdroid/ui/OpenHABWidgetAdapter.java

```diff
@@ -490,11 +493,15 @@ public void bind(OpenHABWidget widget) {
                 } else {
                     progress = widget.getItem().getStateAsFloat().intValue();
                 }
-                mSeekBar.setProgress(progress);
             } else {
-                mSeekBar.setProgress(0);
+                progress = 0;
             }
+            mSeekBar.setProgress(progress);
             mBoundItem = widget.getItem();
+            if(widget.getSwitchSupport()) {
+                mSliderSwitch.setChecked(progress != 0 ? true : false);
+                mSliderSwitch.setVisibility(View.VISIBLE);
+            }
```

Comment:
... else set visibility GONE.

Reply:
I set the visibility to gone in the xml. Do I need to change it here?

Reply:
Yes, as the holder may be re-used for a widget not supporting the switch when scrolling the list.

Reply:
Done